service/src/main/kotlin/fi/espoo/evaka/shared/message/MessageProvider.kt

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.shared.message
+
+interface IMessageProvider {
+    fun get(type: MessageType, lang: MessageLanguage): String
```

Comment:
Is there some advantage to using one function + enum with N variants + when checks, vs just using N functions? The enum solution seems more complicated to me.

In other words:

```kotlin
enum class MessageType {
  A,
  B
}

fun get(type: MessageType, lang: MessageLanguage) {
  return when(type) {
    MessageType.A -> // ...
    MessageType.B -> // ...    
  }
}
```

vs

```kotlin
fun getA(lang: MessageLanguage) {
  // ...
}
fun getB(lang: MessageLanguage) {
  // ...
}
```

Reply:
Yeah, I think you're right. I'll change this

Reply:
Done